java/src/main/native/src/row_conversion.cu

```diff
@@ -1681,26 +1789,74 @@ convert_to_rows(table_view const &tbl, batch_data &batch_info, offsetFunctor off
   dim3 blocks(util::div_rounding_up_unsafe(gpu_tile_infos.size(), NUM_TILES_PER_KERNEL_TO_ROWS));
   dim3 threads(256);
 
+  // build validity tiles for ALL columns, variable and fixed width.
   auto validity_tile_infos = detail::build_validity_tile_infos(
-      num_columns, num_rows, shmem_limit_per_tile, batch_info.row_batches);
+      tbl.num_columns(), num_rows, shmem_limit_per_tile, batch_info.row_batches);
 
   auto dev_validity_tile_infos = make_device_uvector_async(validity_tile_infos, stream);
   dim3 validity_blocks(
       util::div_rounding_up_unsafe(validity_tile_infos.size(), NUM_VALIDITY_TILES_PER_KERNEL));
   dim3 validity_threads(std::min(validity_tile_infos.size() * 32, 128lu));
 
+  auto const validity_offset = column_info.fixed_width_column_starts.back();
+
   detail::copy_to_rows<<<blocks, threads, total_shmem_in_bytes, stream.value()>>>(
-      num_rows, num_columns, shmem_limit_per_tile, gpu_tile_infos, dev_input_data.data(),
-      dev_col_sizes.data(), dev_col_starts.data(), offset_functor,
+      num_rows, num_fixed_width_columns, shmem_limit_per_tile, gpu_tile_infos,
+      dev_input_data.data(), dev_col_sizes.data(), dev_col_starts.data(), offset_functor,
       batch_info.d_batch_row_boundaries.data(),
       reinterpret_cast<int8_t **>(dev_output_data.data()));
 
+  // note that validity gets the entire table and not the fixed-width portion
   detail::copy_validity_to_rows<<<validity_blocks, validity_threads, total_shmem_in_bytes,
                                   stream.value()>>>(
-      num_rows, num_columns, shmem_limit_per_tile, offset_functor,
-      batch_info.d_batch_row_boundaries.data(), dev_output_data.data(), column_starts.back(),
+      num_rows, tbl.num_columns(), shmem_limit_per_tile, offset_functor,
+      batch_info.d_batch_row_boundaries.data(), dev_output_data.data(), validity_offset,
       dev_validity_tile_infos, dev_input_nm.data());
 
+  if (!fixed_width_only) {
+    // build table view for variable-width data only
+    auto variable_width_table = [&tbl]() {
```

Comment:
If we were to move the commonality between this lambda and its [previous occurrence](https://github.com/rapidsai/cudf/pull/10235/files#diff-3bb26954174e53ccc1347458651ac09d63c14a83ba7b71b53b6328947fdea1d7R1723) to a common spot, would that be ~more readable~ less verbose?

```c++
auto select_columns = []( auto const& tbl, auto column_predicate ) {
      std::vector<size_type> indices;
￼      for (int i = 0; i < tbl.num_columns(); ++i) {
￼        if (column_predicate(tbl.column(i))) {
￼          indices.push_back(i);
￼        }
￼      }
￼      return tbl.select(indices);
};
```

Reply:
That link should have been to: https://github.com/rapidsai/cudf/pull/10235/files#diff-3bb26954174e53ccc1347458651ac09d63c14a83ba7b71b53b6328947fdea1d7R1777

Reply:
Again, please feel free to ignore, if eliminating this repetition doesn't save us anything.

Reply:
Not at all. It's a solid point and reduces duplication, which can only reduce bug creep.